### PR TITLE
Add IP Address to gui debug tab

### PR DESCRIPTION
**Pull request by InvincibleRMC**
# Pull Request

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Adds ip address of pi to debug tab of gui.

## Related Tickets & Documents
#71

### CODE REVIEW

**Review by NoahMollerstuen — APPROVED** [review]
Looks good to me. Only thing I would say is that we should merge the widget into the right bar with thruster testing and arming, and the bar down into the console's space, but we also need to reorganize everything else in the guis so that should probably be a separate issue.

File: src/surface/gui/gui/pilot_app.py [hunk 1]
```diff
@@ -12,25 +12,27 @@ def __init__(self) -> None:
...
-
-        self.floodWidget: FloodWarning = FloodWarning()
-        layout.addWidget(self.floodWidget, alignment=Qt.AlignmentFlag.AlignRight
-                         | Qt.AlignmentFlag.AlignTop)
-
-        self.arm: Arm = Arm()
-        layout.addWidget(self.arm,
-                         alignment=Qt.AlignmentFlag.AlignRight | Qt.AlignmentFlag.AlignBottom)
+        # Look into QStackedLayout for possibly switching between
+        # 1 big camera feed and 2 smaller ones
```
Comment (benjaminwp18) [inline src/surface/gui/gui/pilot_app.py:19]: Mark with TODO keyword or add to git issue instead

File: src/surface/gui/gui/widgets/ip_widget.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+from rov_msgs.msg import IPAddress
+
+
+class IPWidget(QWidget):
+
+    signal = pyqtSignal(IPAddress)
+
+    def __init__(self) -> None:
+        super().__init__()
+        # Boilerplate PyQt Setup to link to ROS through a signal/subscriber
```
Comment (benjaminwp18) [inline src/surface/gui/gui/widgets/ip_widget.py:14]: why is comment?
rm comment
unless you want to make a ticket to abstract this

File: src/surface/gui/gui/widgets/ip_widget.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        self.signal.connect(self.refresh)
+        self.sub = GUIEventSubscriber(IPAddress, 'ip_address', self.signal)
+
+        ip_layout = QVBoxLayout()
+        self.ethernet_label = QLabel(f'Ethernet: {IPAddress.ETHERNET_ADDRESS__DEFAULT}')
+        self.wireless_label = QLabel(f'Wireless: {IPAddress.WIRELESS_ADDRESS__DEFAULT}')
```
Comment (benjaminwp18) [inline src/surface/gui/gui/widgets/ip_widget.py:20]: WHAT IS THIS DARK MAGICK???
Are these supposed to be exposed/used?

File: src/surface/gui/launch/operator_launch.py [hunk 4]
```diff
@@ -18,7 +18,8 @@ def generate_launch_description() -> LaunchDescription:
                     ("/surface/gui/task_feedback", "/surface/task_feedback"),
                     ("/surface/gui/auto_docker_control", "/surface/auto_docker_control"),
                     ("/surface/gui/vehicle_state_event", "/surface/vehicle_state_event"),
-                    ("/surface/gui/mavros/cmd/arming", "/tether/mavros/cmd/arming")],
+                    ("/surface/gui/mavros/cmd/arming", "/tether/mavros/cmd/arming"),
+                    ("/surface/gui/ip_address", '/tether/ip_address')],
```
Comment (benjaminwp18) [inline src/surface/gui/launch/operator_launch.py:22]: double/single quote consistency

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]
oh my goodness gracious noah's spelling is bad

File: src/surface/gui/gui/widgets/ip_widget.py (hunk 3 above)
Comment (InvincibleRMC) [reply src/surface/gui/gui/widgets/ip_widget.py:20]: yes I set a default

**Review by benjaminwp18 — APPROVED** [review]

**Pull request merged**